packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_services.dart

```diff
@@ -205,9 +210,22 @@ class DeepLinksService {
                   ?.cast<Map<String, Object?>>();
               if (aasaPaths != null) {
                 for (final aasaPath in aasaPaths) {
-                  paths
-                      .putIfAbsent(domainName, () => <String>[])
-                      .add(aasaPath[_pathKey] as String);
+                  final rawQueryParams = (aasaPath[_queryParamsKey] as List?)
+                      ?.cast<Map<String, Object?>>();
+                  final Map<String, String> queryParams = rawQueryParams != null
+                      ? {
+                          for (final item in rawQueryParams)
+                            item[_keyKey] as String: item[_valueKey] as String,
+                        }
+                      : {};
```

Comment:
this can be simplified:
```
final queryParams = <String, String>{
  for (final item in rawQueryParams ?? [])
    item[_keyKey] as String: item[_valueKey] as String,
};
```